mlos_bench/mlos_bench/optimizers/mlos_core_optimizer.py

```diff
@@ -38,6 +40,23 @@ def __init__(self, tunables: TunableGroups, service: Optional[Service], config:
         opt_type = getattr(OptimizerType, self._config.pop(
             'optimizer_type', DEFAULT_OPTIMIZER_TYPE.name))
 
+        if opt_type == OptimizerType.SMAC:
+            # If output_directory is specified, turn it into an absolute path.
+            output_directory = self._config.get('output_directory')
+            if output_directory is not None:
+                if not os.path.isabs(output_directory):
+                    self._config['output_directory'] = os.path.join(os.getcwd(), output_directory)
+            else:
+                _LOG.warning("No output_directory specified. SMAC will use temporary directory.")
+
+            # Set max_trials == max_iterations.
+            if 'max_trials' not in self._config:
+                self._config['max_trials'] = self._max_iter
+            assert self._config['max_trials'] == self._max_iter, \
+                f"max_trials {self._config['max_trials']} != max_iterations {self._max_iter}"
+
+            # TODO: Set run_name from experimentId (global_params not currently accessible from here).
```

Comment:
Would like to add support for this to make examining logs easier, but it would require some changes to the `__init__` signature that I haven't dug through yet.

Reply:
#474 is needed before this